src/scipp/core/bins.py

```diff
@@ -114,6 +114,11 @@ def lookup(
         dims = (*[d for d in func.dims if d != dim], dim)
         func.data = func.data.transpose(dims).copy()
         func.coords[dim] = func.coords[dim].transpose(dims).copy()
+        for key, mask in func.masks.items():
+            func.masks[key] = mask.transpose(
+                # Masks potentially have fewer dims than the data.
+                [d for d in dims if d in mask.dims] or None
+            ).copy()
```

Comment:
Was this actually required, i.e., did the new test raise without this?

Reply:
Yes it did raise.